processors/processor_factory.go

```diff
@@ -24,14 +25,23 @@ func (f ProcessorConstructorFunc) New() Processor {
 // This layer of indirection allows for different processor integrations to be compiled in or compiled out by `go build --tags ...`
 var processorImpls = make(map[string]ProcessorConstructor)
 
+// processorMetaData is a k/v store from processor names to their sample metadata
+var processorMetaData = make(map[string]ProcessorMetadata)
+
 // RegisterProcessor is used to register ProcessorConstructor implementations. This mechanism allows
 // for loose coupling between the configuration and the implementation. It is extremely similar to the way sql.DB
 // driver's are configured and used.
 func RegisterProcessor(name string, constructor ProcessorConstructor) error {
 	if _, ok := processorImpls[name]; ok {
 		return fmt.Errorf("processor already exists")
 	}
+
+	if _, ok := processorMetaData[name]; ok {
```

Comment:
I don't think this can happen. if name already exists, it should've exited in line 36?

Reply:
It's more of a belt and suspenders thing